### PR TITLE
[JENKINS-58028] Add GMavenPlus version

### PR DESCRIPTION
**Pull request by bitwiseman**
Eventually will replace gmaven with gmavenplus.

This commit should not change downstream behavior aside from setting the version of GMavenPlus.

### CODE REVIEW

**Comment by bitwiseman** [comment]
@oleg-nenashev  
See https://github.com/jenkinsci/plugin-pom/pull/209 and related for examples.  
However, there's no downstream changes needed for this change. This only _enables_ use of gmavenplus with a consistent version.   I'm not even sure what projects are directly downstream from this pom.  Suggestions?

**Comment by oleg-nenashev** [comment]
Right, I did not notice that the execution section was for `lifecycleMappingMetadata`

**Comment by oleg-nenashev** [comment]
Any plans to finish it @bitwiseman ?

**Comment by bitwiseman** [comment]
@oleg-nenashev  I've been heads down on Matrix.  I'll get back to it when that clears.

**Comment by oleg-nenashev** [comment]
Should we go ahead and replace GMaven @jenkinsci/core ? AFAICT it is no longer maintained, and the same patch has worked quite well for plugins

**Comment by jglick** [comment]
Merge conflicts here.

This is OK though I would still generally recommend components not use Groovy sources to begin with. `pipeline-model-definition` is the main offender (among Maven-based repos at least).

**Comment by basil** [comment]
Do we need this in the parent POM? I looked around for usages of GMaven in any POMs whose parent is this repository and the only two I could find were [`acceptance-test-harness`](https://github.com/jenkinsci/acceptance-test-harness/blob/070053a9122c52dac16231e5abc3b6ea25b05b19/pom.xml#L498-L522) and [`lib-groovy-guice-binder`](https://github.com/jenkinsci/lib-groovy-guice-binder/blob/ed3447b4ed59aa487cda6690072211c1678de3c6/pom.xml#L16-L30) (itself only consumed by `acceptance-test-harness`). If this is only used by `acceptance-test-harness` then perhaps we should close this PR and just migrate `acceptance-test-harness` and `lib-groovy-guice-binder` to GMavenPlus directly?

**Comment by bitwiseman** [comment]
@basil  
Having this version controlled in the POM would be better.  

As Oleg noted GMaven is no longer maintained and this replaces that.  If anyone uses GMaven, this makes it easy for them to switch and standardized the version of GMavenPlus that gets used.

File: pom.xml [hunk 1]
```diff
@@ -957,7 +979,8 @@
     <cobertura-maven-plugin.version>2.7</cobertura-maven-plugin.version>
     <exec-maven-plugin.version>3.0.0</exec-maven-plugin.version>
     <gmaven-plugin.version>1.5-jenkins-3</gmaven-plugin.version>
-    <gwt-maven-plugin.version>2.9.0</gwt-maven-plugin.version>
+    <gmavenplus-plugin.version>1.7.1</gmavenplus-plugin.version>
+    <gwt-maven-plugin.version>2.8.2</gwt-maven-plugin.version>
```
Comment (bitwiseman) [inline pom.xml:983]:
```suggestion
    <gmavenplus-plugin.version>1.13.1</gmavenplus-plugin.version>
    <gwt-maven-plugin.version>2.9.0</gwt-maven-plugin.version>
```

**Comment by basil** [comment]
> Having this version controlled in the POM would be better.

I trust that you've looked into this, but I don't quite see the reason why it would be better. From a different perspective, it could be argued that slimming down the dependency tree of the POM would be better. Either way, I have no strong preference, but I do have a preference that we file downstream PRs in `acceptance-test-harness`](https://github.com/jenkinsci/acceptance-test-harness/blob/070053a9122c52dac16231e5abc3b6ea25b05b19/pom.xml#L498-L522) and [`lib-groovy-guice-binder`] adapting them to the changes in this PR.

**Review by jglick — COMMENTED** [review]
-0 I suppose. Harmless, though most components should not be using Groovy sources, and for those that do, I see no particular advantage in managing the mojo version here.

**Comment by bitwiseman** [comment]
> -0 I suppose. Harmless, though most components should not be using Groovy sources, and for those that do, I see no particular advantage in managing the mojo version here.

For the projects that do, it is better that we guide them away from GMaven to GMavenPlus and to not managing the version of the plugin on their own.

**Comment by jglick** [comment]
> guide them away from GMaven to GMavenPlus

Sure.

> to not managing the version of the plugin on their own

Why? Usually we add a Maven plugin version to a POM either because it is widely used, or because we define some configuration for it, or because unexpected updates could have bad interactions with something else we define. Miscellaneous plugins used by particular components for idiosyncratic reasons (Shade, for example) do not particularly benefit from being defined in some parent. It just makes for one extra level of indirection in Dependabot.

**Comment by jglick** [comment]
As of https://github.com/jenkinsci/acceptance-test-harness/pull/719 we are down to `lib-groovy-guice-binder`, an odd little library which seems to be a dependency of `acceptance-test-harness`.

**Comment by timja** [comment]
Apparently used for https://github.com/jenkinsci/acceptance-test-harness/blob/master/docs/WIRING.md#configuring-how-to-run-tests

@jtnord / @olivergondza are either of you using this functionality?

**Comment by jglick** [comment]
Yes various ATH scripts define `CONFIG` as per [this implementation](https://github.com/jenkinsci/acceptance-test-harness/blob/a9027d40be247fbc51f4ea4af7e4bd0bd7f7e6e8/src/main/java/org/jenkinsci/test/acceptance/Config.java#L18-L19); [example](https://github.com/jenkinsci/acceptance-test-harness/blob/a9027d40be247fbc51f4ea4af7e4bd0bd7f7e6e8/pom.xml#L670).

**Comment by basil** [comment]
There has been no interest in this during the intervening year.

**Comment by jglick** [comment]
In fact we have been moving to remove `src/{main,test}/groovy/` in various repos.

**Pull request closed without merging**